store/snapshots/store.go

```diff
@@ -172,14 +173,20 @@ func (s *Store) Load(height uint64, format uint32) (*types.Snapshot, <-chan io.R
 				_ = pw.CloseWithError(err)
 				return
 			}
-			defer chunk.Close()
-			_, err = io.Copy(pw, chunk)
+			err = func() error {
+				defer chunk.Close()
+				if _, err := io.Copy(pw, chunk); err != nil {
+					return fmt.Errorf("failed to copy chunk %d: %w", i, err)
+				}
+				if err != nil {
+					_ = pw.CloseWithError(err)
+					return err
+				}
```

Comment:
From the original code and we can adapt it in here
```go
defer chunk.Close()

if _, err := io.Copy(pw, chunk); err != nil {
      _ = pw.CloseWithError(err)
      return fmt.Errorf("failed to copy chunk %d: %w", i, err)
}

return pw.Close()
```

Reply:
Thanks for your support @odeke-em. Very appreciated!

Reply:
Cheers, thanks for the fixes!